plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestingExasolServer.java

```diff
@@ -0,0 +1,156 @@
+import com.exasol.containers.ExasolService;
+import com.exasol.jdbc.EXADriver;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.jdbc.BaseJdbcConfig;
+import io.trino.plugin.jdbc.ConnectionFactory;
+import io.trino.plugin.jdbc.DriverConnectionFactory;
+import io.trino.plugin.jdbc.RetryingConnectionFactory;
+import io.trino.plugin.jdbc.credential.StaticCredentialProvider;
+import io.trino.testing.ResourcePresence;
+
+import java.io.Closeable;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.testing.TestingConnectorSession.SESSION;
+import static java.lang.String.format;
+
+public class TestingExasolServer
+        implements Closeable
+{
+    public static final String TEST_USER = "trino_test";
+    /**
+     * Name of the test schema. Must not contain an underscore, required by {@link io.trino.plugin.exasol.TestExasolConnectorTest#testShowSchemasLikeWithEscape()}
+     */
+    public static final String TEST_SCHEMA = "trinotestschema";
+    public static final String TEST_PASS = "trino_test_password";
+
+    private final ExasolContainer<?> container;
+
+    private TestingExasolServer()
+    {
+        container = startContainer();
+    }
+
+    private static ExasolContainer<?> startContainer()
+    {
+        ExasolContainer<?> container = new ExasolContainer<>().withRequiredServices(ExasolService.JDBC);
+        // Deactivate reuse to allow running test classes in parallel.
+        container.withReuse(false);
+        container.start();
+        return container;
+    }
+
+    public static TestingExasolServer start()
+    {
+        TestingExasolServer server = new TestingExasolServer();
+        server.createUserAndSchema();
+        return server;
+    }
+
+    private void createUserAndSchema()
+    {
+        executeAsSys(List.of(format("CREATE USER %s IDENTIFIED BY \"%s\"", TEST_USER, TEST_PASS),
+                format("GRANT CREATE SESSION TO %s", TEST_USER),
+                format("GRANT CREATE SCHEMA TO %s", TEST_USER),
+                format("GRANT CREATE TABLE TO %s", TEST_USER),
+                format("GRANT IMPORT TO %s", TEST_USER),
+                format("GRANT CREATE VIEW TO %s", TEST_USER),
+                format("GRANT CREATE ANY VIEW TO %s", TEST_USER)));
+        execute(List.of(format("CREATE SCHEMA %s", TEST_SCHEMA)), TEST_USER, TEST_PASS);
+    }
+
+    public String getJdbcUrl()
+    {
+        return container.getJdbcUrl();
+    }
+
+    public void execute(String sql)
+    {
+        execute(sql, TEST_USER, TEST_PASS);
+    }
+
+    private void executeAsSys(List<String> sql)
+    {
+        execute(sql, container.getUsername(), container.getPassword());
+    }
+
+    public void execute(String sql, String user, String password)
+    {
+        execute(List.of(sql), user, password);
+    }
+
+    public void execute(List<String> statements, String user, String password)
```

Comment:
We don't need `List<String> statements`. Having a method with `String sql` is sufficient.

Reply:
✅